RecoEgamma/EgammaElectronProducers/src/LowPtGsfElectronFeatures.cc

```diff
@@ -165,78 +165,460 @@ namespace lowptgsfeleid {
     }
 
     // GSF tracks
-    if (ele->core().isNonnull()) {
-      const auto& gsf = ele->core()->gsfTrack();  // reco::GsfTrackRef
+    if (ele.core().isNonnull()) {
+      reco::GsfTrackRef gsf = ele.core()->gsfTrack();
       if (gsf.isNonnull()) {
         gsf_mode_p = gsf->pMode();
         eid_gsf_nhits = (float)gsf->found();
         eid_gsf_chi2red = gsf->normalizedChi2();
         TVector3 gsfTV3(0, 0, 0);
         gsfTV3.SetPtEtaPhi(gsf->ptMode(), gsf->etaMode(), gsf->phiMode());
         TVector3 eleTV3(0, 0, 0);
-        eleTV3.SetPtEtaPhi(ele->pt(), ele->eta(), ele->phi());
+        eleTV3.SetPtEtaPhi(ele.pt(), ele.eta(), ele.phi());
         gsf_dr = eleTV3.DeltaR(gsfTV3);
       }
     }
 
     // Super clusters
-    if (ele->core().isNonnull()) {
-      const auto& sc = ele->core()->superCluster();  // reco::SuperClusterRef
+    if (ele.core().isNonnull()) {
+      reco::SuperClusterRef sc = ele.core()->superCluster();
       if (sc.isNonnull()) {
         eid_sc_E = sc->energy();
         eid_sc_eta = sc->eta();
         eid_sc_etaWidth = sc->etaWidth();
         eid_sc_phiWidth = sc->phiWidth();
-        sc_Nclus = (float)sc->clustersSize();
+        sc_Nclus = sc->clustersSize();
       }
     }
 
     // Track-cluster matching
-    if (ele.isNonnull()) {
-      eid_match_seed_dEta = ele->deltaEtaSeedClusterTrackAtCalo();
-      eid_match_eclu_EoverP = (1. / ele->ecalEnergy()) - (1. / ele->p());
-      eid_match_SC_EoverP = ele->eSuperClusterOverP();
-      eid_match_SC_dEta = ele->deltaEtaSuperClusterTrackAtVtx();
-      eid_match_SC_dPhi = ele->deltaPhiSuperClusterTrackAtVtx();
-    }
+    //if (ele.isNonnull()) {
+    eid_match_seed_dEta = ele.deltaEtaSeedClusterTrackAtCalo();
+    eid_match_eclu_EoverP = (1. / ele.ecalEnergy()) - (1. / ele.p());
+    eid_match_SC_EoverP = ele.eSuperClusterOverP();
+    eid_match_SC_dEta = ele.deltaEtaSuperClusterTrackAtVtx();
+    eid_match_SC_dPhi = ele.deltaPhiSuperClusterTrackAtVtx();
+    //}
 
     // Shower shape vars
-    if (ele.isNonnull()) {
-      eid_shape_full5x5_HoverE = ele->full5x5_hcalOverEcal();
-      eid_shape_full5x5_r9 = ele->full5x5_r9();
-    }
+    //if (ele.isNonnull()) {
+    eid_shape_full5x5_HoverE = ele.full5x5_hcalOverEcal();
+    eid_shape_full5x5_r9 = ele.full5x5_r9();
+    //}
 
     // Misc
     eid_rho = rho;
 
-    if (ele.isNonnull()) {
-      eid_brem_frac = ele->fbrem();
-      core_shFracHits = (float)ele->shFracInnerHits();
+    //if (ele.isNonnull()) {
+    eid_brem_frac = ele.fbrem();
+    core_shFracHits = ele.shFracInnerHits();
+    //}
+
+    // Unbiased BDT from ElectronSeed
+    gsf_bdtout1 = unbiased;
+
+    // Clusters
+    if (ele.core().isNonnull()) {
+      reco::GsfTrackRef gsf = ele.core()->gsfTrack();
+      if (gsf.isNonnull()) {
+        reco::SuperClusterRef sc = ele.core()->superCluster();
+        if (sc.isNonnull()) {
+          // Propagate electron track to ECAL surface
+          double mass_ = 0.000511 * 0.000511;
```

Comment:
```suggestion
          double mass2 = 0.000511 * 0.000511;
```
trailing underscore is reserved for data members

Reply:
Done.